src/equations/compressible_euler_2d.jl

```diff
@@ -384,6 +384,35 @@ Should be used together with [`StructuredMesh`](@ref).
     return boundary_flux
 end
 
+"""
+    get_boundary_outer_state(u_inner, t,
+                             boundary_condition::typeof(boundary_condition_slip_wall),
+                             normal_direction::AbstractVector,
+                             mesh, equations::CompressibleEulerEquations2D,
+                             dg, cache, indices...)
+For subcell limiting, the calculation of local bounds for non-periodic domains requires the boundary
+outer state. This function returns the boundary value for [`boundary_condition_slip_wall`](@ref) at
+time `t` and for node with spatial indices `indices` and at the boundary with `normal_direction`.
+
+Should be used together with [`P4estMesh`](@ref).
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+@inline function get_boundary_outer_state(u_inner, t,
+                                          boundary_condition::typeof(boundary_condition_slip_wall),
+                                          normal_direction::AbstractVector,
+                                          mesh, equations::CompressibleEulerEquations2D,
+                                          dg, cache, indices...)
```

Comment:
Why does this contain `mesh`, `dg`, `cache`, `indices` if they are not used? It is also an (somewhat) ugly mixing of physics and numerical methods that we generally try to avoid...

Reply:
In the [supersonic flow elixir](https://github.com/trixi-framework/Trixi.jl/blob/038b3ca067070a6dfa761d34ce88155d0ab32a4d/examples/p4est_2d_dgsem/elixir_euler_supersonic_cylinder_sc_subcell.jl#L50-L77), this routine is used two more times.
There, I need `cache`, `dg` and `indices` to get the spatial coordinates with `get_node_coords`.

When I look at it again right now, it seems that I don't need the `mesh` parameter. It's from an old state where I needed to distinguish between StructuredMesh and P4estMesh. This works now automatically because there is now parameter `direction for the P4estMesh. I will remove it and test if everything still works. Thanks!

I know, the whole concept of outer states used in the volume integral is ugly. But I don't see another way here :see_no_evil:

Reply:
> When I look at it again right now, it seems that I don't need the mesh parameter. [...] I will remove it and test if everything still works.

It seems to work:+1:

Thinking about `get_node_coords`, I could also remove `dg` when using a `view` instead of `get_node_coords`. But I'm not sure whether that's better. What do you think @sloede?

Reply:
Removing `mesh`is done in [654c86c](https://github.com/trixi-framework/Trixi.jl/pull/1954/commits/654c86c860a4101f06314c81d3c85a658d1a5993).

Reply:
Hm. Would it be a solution to also add the slip wall version to the elixirs - at least for now? It seems to me that there is a better interface design to be found somewhere; I'm just not deep enough into the call graph to figure it out.

Reply:
Of course, I can do that.
Just want to add that there is another version for the Dirichlet boundary condition located in [`dg_2d_subcell_limiting.jl`](https://github.com/trixi-framework/Trixi.jl/blob/90eebb032c23d38b435735a27498a5446ae51a89/src/solvers/dgsem_tree/dg_2d_subcell_limiters.jl#L468-L486). But since this file is specifically for subcell limiting, it's probably okay there.
I cannot put the new slip wall version there since it's for Euler only.

Reply:
Maybe this could then be a good intermediate way forward? I am not against including this implementation in Trixi.jl per se, but at the moment it seems to me that there is still some room for improving the interface. Maybe with some more experience in using these methods this will improve over time, and then we can revisit this discussion.

Maybe just in case you create an issue for this, such that we do not forget it?

Reply:
Yes :+1: I pushed the changes in [6ecc6d7](https://github.com/trixi-framework/Trixi.jl/pull/1954/commits/6ecc6d768556d8d3be17b5738b0f5ab80a96ca7e) and added the issue https://github.com/trixi-framework/Trixi.jl/issues/2053.